freud/diffraction.pyx

```diff
@@ -21,13 +21,220 @@ import scipy.ndimage
 import freud.locality
 
 cimport numpy as np
+from cython.operator cimport dereference
 from libcpp cimport bool as cbool
 
+cimport freud._diffraction
+cimport freud.locality
 cimport freud.util
-from freud.util cimport _Compute
+from freud.locality cimport _SpatialHistogram1D
+from freud.util cimport _Compute, vec3
 
 logger = logging.getLogger(__name__)
 
+cdef class StaticStructureFactorDebye(_Compute):
+    r"""Computes a 1D static structure factor using Debye scattering equation.
+
+    This computes the static `structure factor
+    <https://en.wikipedia.org/wiki/Structure_factor>`__ :math:`S(k)`, assuming
+    an isotropic system (averaging over all :math:`k` vectors of the same
+    magnitude). This is implemented using the Debye scattering equation:
+
+    .. math::
+
+        S(k) = \frac{1}{N} \sum_{i=0}^{N} \sum_{j=0}^{N} \text{sinc}(k r_{ij})
+
+    where :math:`N` is the number of particles, :math:`\text{sinc}` function is
+    defined as :math:`\sin x / x` (no factor of :math:`\pi` as in some
+    conventions). For more information see `here
+    <https://en.wikipedia.org/wiki/Structure_factor>`__. The equation 4 from
+    the link can be obtained by replacing :math:`\frac{\sin(k r)}{kr}` with
+    :math:`\text{sinc}(k r)`.
+
+    .. note::
+        This code assumes all particles have a form factor :math:`f` of 1.
+
+    Partial structure factors can be computed by providing ``query_points`` to the
+    :py:meth:`compute` method. When computing a partial structure factor, the
+    total number of points in the system must be specified. The normalization
+    criterion is based on the Faber-Ziman formalism. For particle types
+    :math:`\alpha` and :math:`\beta`, we compute the total scattering function
+    as a sum of the partial scattering functions as:
```

Comment:
According to the description in the PR, `query_points` has to be the same as `points`. Is that outdated now?

Reply:
This is correct. Multi-species structure factors (or partial structure factors) were originally to be added later, but were added as implementation wasn't to hard. The docs are correct.

Reply:
We have implemented the support for "multi-species structure factors" (partial structure factors). This can be done by providing different set of `query_points` and `points` along with `N_total`. The original PR description is outdated.